### ISSUE TITLE
object-shorthand not being enforced when using "avoidQuotes"

### ISSUE THREAD

**Issue by TheSavior** [eslint/eslint#6429]
**What version of ESLint are you using?**
2.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
either
```
"object-shorthand": ["error", "always", {"avoidQuotes": true}],
```
or
```
/*eslint object-shorthand: ["error", "always", { "avoidQuotes": true }]*/
```

**What did you do? Please include the actual source code causing the issue.**
Object shorthand is not being enforced. We configured the rule in our eslintrc and have an object block that should fail. As a minimal test case I have created a repo with a single file and configures the rule inline in the file. That should definitely cause it to fail, but `npm test` passes.

https://github.com/TheSavior/eslint-shorthand-testcase

I'm sure this is user error since the eslint test suite is passing, but we haven't been able to figure out what we are doing wrong with this rule. Other rules are working for us no problem.

**Comment by ilyavolodin** [comment]
Verified in online demo. Both of the examples below return no error:
```js
/*eslint object-shorthand: ["error", "always", {"avoidQuotes": true}]*/
/*eslint-env es6*/

var foo = {
    w: {}
};
```

```js
/*eslint object-shorthand: ["error", "always", { "avoidQuotes": true }]*/
/*eslint-env es6*/

var foo = {
    w() {}
};
```

**Comment by TheSavior** [comment]
Good to know that I'm not crazy. This has been occurring since at least `2.11.1`.

**Comment by ilyavolodin** [comment]
Btw, the rule works fine without `avoidQuotes` option. I think there's some issue with that option alone.

**Comment by mysticatea** [comment]
I'm working on this.
